File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/splash/SplashModule.java
package com.github.alexeygorovoy.moxytemplate.dagger.demo.splash;

import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.presenter.SplashPresenter;
import com.github.alexeygorovoy.moxytemplate.rx.RxSchedulers;

import dagger.Module;
import dagger.Provides;


@Module
public class SplashModule {

    @Provides
    SplashPresenter provideSplashPresenter(RxSchedulers schedulers) {
        return new SplashPresenter(schedulers);
    }

}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/splash/SplashModule.kt
package com.github.alexeygorovoy.moxytemplate.dagger.demo.splash

import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.presenter.SplashPresenter
import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.presenter.SplashPresenterImpl
import dagger.Module
import dagger.Provides

@Module
class SplashModule {

    @Provides
    internal fun provideSplashPresenter(presenter: SplashPresenterImpl): SplashPresenter = presenter
}

